### PR TITLE
Provide RefreshIn for app token provider

### LINKED ISSUE #4074: [Feature Request] Infer refersh_in when using AppTokenProvider
https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2522531

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #4074

Related to https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4062 but for the AppTokenProvider

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj [hunk 1]
```diff
@@ -31,8 +31,8 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.2.0" />
-    <PackageReference Include="MSTest.TestAdapter" Version="2.2.10" />
-    <PackageReference Include="MSTest.TestFramework" Version="2.2.10" />
+    <PackageReference Include="MSTest.TestAdapter" Version="3.0.2" />
+    <PackageReference Include="MSTest.TestFramework" Version="3.0.2" />
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj:35]: My VS refuses to run with the old values :(

File: tests/Microsoft.Identity.Test.Unit/ResourceHelper.cs [hunk 2]
```diff
@@ -18,12 +18,7 @@ public class ResourceHelper
         /// </remarks>
         public static string GetTestResourceRelativePath(string resourceName)
         {
-
-#if DESKTOP
```
Comment (neha-bhargava) [inline tests/Microsoft.Identity.Test.Unit/ResourceHelper.cs:22]: Is this check no longer needed?

**Review by neha-bhargava — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Unit/AppTokenProviderTests.cs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                                                    .ExecuteAsync(new CancellationToken()).ConfigureAwait(false);
+
+            Assert.IsNotNull(result.AccessToken);
+            Assert.AreEqual(TestConstants.DefaultAccessToken + differentScopesForAt, result.AccessToken);
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+            Assert.AreEqual(4, callbackInvoked);
+        }
+
+        [TestMethod]
+        public async Task CheckRefreshInAsync()
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/AppTokenProviderTests.cs:108]: - Missing test case when AppTokenProviderResult.RefreshInSeconds is set.
- Can this be refactored to use DataRows like [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/8a0a64008bb6053cde73a48bf8f8c7ae8faf4664/tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ManagedIdentityTests.cs#L447-L452)? Maybe pass in expiration time and expected RefreshOn.HasValue

**Review by pmaytak — CHANGES_REQUESTED** [review]
Has bug, I think?

File: tests/Microsoft.Identity.Test.Unit/AppTokenProviderTests.cs [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Extensibility;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit
+{
+    [TestClass]
+    public class AppTokenProviderTests : TestBase
```
[the remark this answers is not in the document]
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Unit/AppTokenProviderTests.cs:19]: We did, I moved it to a separate file and add a new test. I should have mentioned it in the pr

File: tests/Microsoft.Identity.Test.Unit/ResourceHelper.cs (hunk 2 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Unit/ResourceHelper.cs:22]: Looks like the updated the mstest dependencies "fixed" this inconsistency between .NET and .NET Fwk

**Review by pmaytak — APPROVED** [review]

**Pull request merged**